content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+> details about how and why we reach to the conclusion that that is a valid and desirable solution for the
+> problem at hand
+
+There are diverse situation in which we want to create an NPM package that contains
+a [React][1] Component with css and images.
+
+This situation is particularly common inside companies that want to share components,
+and pieces of code in general, between apps usually inside a private NPM repository, but
+it also apply to Open Source software like a single Component package such as a Date Picker
+or a Multi Select and also to Component Libraries or UI Frameworks like [React Bootstrap][9],
+[Ant Design][8], [Semantic UI][4], [Material UI][7], etc (more on these later).
+
+This problem might seem simple enough but it turns out that it has a lot of nuances and trade-offs.
+
+The main challenge in packaging a [React][1] Component is surprisingly not Javascript,
+but all the Assets that might be related to that component such as CSS, Images, SVGs and fonts.
+These assets are directly related to our Javascript code via some of the following statements:
+
+- `imports image from './image.{png,jpg,svg}'` in Javascript
+- `import classNames from './styles.{css,scss,less)'` in Javascript
+- `url(image.{png,jpg,svg})` in CSS
+- `font-family and @import` in CSS
+
+[Webpack][2] can understand all these dependency statements via _loaders_ and completely
+analyse the cross-language and cross-file-type dependency graph, which for a very simple
+component might look something like this.
+
+```
+            --> styles.css --> otherImage.svg
+MyComponent --> image.svg
+
+```
+
+It turns out that there are a lot of schemes for representing this graph in a Packaged NPM module,
+each with it's own trade-offs. Analysing each alternative is the main purpose of this post.
+
+Unfortunately in early 2020 there still no universal standard way of Packaging a Front End module with all
+its assets, but a lot of progress has been made and there are actually lots of promising solutions.
+
+We are going to explore how this problem is tackled in the wild by studying
+each technique and providing example real world production ready libraries that
+use them and in the end of this post we are going
+to recommend a particular solution to this problem based of my experience.
+
+Additionally we are going to explore how each of these packaging schemes enable
+different **Theming** schemes and, at the end, recommend one.
+
+Most of this post is also applicable to any piece of Front End Javascript code that is related
+to CSS, images, etc, such as [Angular][3] Components / Directives, [Vue][15] Components, etc.
+
+## To Inline or not to Inline
+
+In the context of Front End assets, inlining means to bake those assets directly
+into the Javascript code (bundle) generated by our bundlers or compilers such as Webpack or [Babel][4].
+
+CSS can be inlined without any third party library but to enable all regular CSS features
```

Comment:
To me it was clear that you can inline CSS but without searching, I did not understand what you meant by "to enable all regular CSS features, a third-party CSS-In-JS library is recommended". Specifically the first part about enabling _regular CSS features_ which sounds a little vague or non-specific. Not saying that's bad, but it made me second-guess myself when I was reading this.

Reply:
nit: Also, there are 4 instances of "more on this later" in the text, 2 after a comma and 2 within parentheses. I prefer when it's in parentheses. Personally I think it can  seem a little awkard after a comma.

Reply:
Ok, so I rephrase it differently, let me know what you think

Reply:
Replaced all occurrences of `more on this later` for `(more on this later)`